test/csuite/Makefile.am

```diff
@@ -57,6 +57,9 @@ noinst_PROGRAMS += test_wt3135_search_near_collator
 test_wt3184_dup_index_collator_SOURCES = wt3184_dup_index_collator/main.c
 noinst_PROGRAMS += test_wt3184_dup_index_collator
 
+test_wt3338_partial_update_SOURCES = wt3338_partial_update/main.c
+noinst_PROGRAMS += test_wt3338_partial_update
+
```

Comment:
There already is a 'compat' option to `test/format`.  Hmm, so it is probably actually generating invalid log files as there likely are transactions in flight when the version changes.  It would be interesting to run printlog over the logs if logging, compatibility and modify are all going on in test/format.

I also don't think WT-3417 is needed for a unit test.  That is only needed in the case where there can be outstanding transactions in progress when the compatibility version changes.  It would be easy to construct a test that doesn't do that.  (And I definitely agree we need a test that does have it and specifically tests the contents of log files.)